jbpm-flow/src/main/java/org/jbpm/process/instance/impl/CancelNodeInstanceAction.java

```diff
@@ -38,7 +40,7 @@ public void execute(ProcessContext context) throws Exception {
 		WorkflowProcessInstance pi = context.getNodeInstance().getProcessInstance();
 		NodeInstance nodeInstance = findNodeByUniqueId(pi.getNodeInstances(), attachedToNodeId);
 		if (nodeInstance != null) {
-		    ((org.jbpm.workflow.instance.NodeInstance)nodeInstance).cancel();
+            ((org.jbpm.workflow.instance.NodeInstance) nodeInstance).cancel(TIMED);
```

Comment:
I think CancelNodeInstanceAction is used in multiple cases, boundary escalation and boundary error events, not sure I grasp the meaning of TIMED cancel type in this context?

Reply:
You are right. I can use SKIPPED vs OBSOLETE. OBSOLETE means the task has no meaning anymore as the group it belonged is already completted. SKIPPED would mean that another route has been taking.